content/modules/redisearch/redisearch-2-upgrade.md

```diff
@@ -0,0 +1,74 @@
+---
+Title: Upgrading a Database to RediSearch 2.x
+description:
+weight: 30
+alwaysopen: false
+categories: ["Modules"]
+module: RediSearch
+---
+RediSearch 2.x includes some significant changes to the architecture of RediSearch that provide improved functionality.
+The main architectural change is that the indexes are stored outside of the Redis database that contains the data.
+This allows for improvements in command efficiency.
+It also allows for improvements in replication between clusters because the index changes are managed by the participating clusters rather than being synchronized with the data.
+
+This change allows databases with RediSearch to support:
+
+- Active-Active databases
+- Database clustering (sharding)
+- Replica Of to sharded destination database
+- EXPIRE of documents reflected in the index
+
+To upgrade a Redis Enterprise Software (RS) database with RediSearch 1.x to RediSearch 2.x, you have to set up a new database with RediSearch 2.x and use the `RediSearch_Syncer.py` script to replicate the data from the old database into the new database.
+
+## Prerequisites
+
+Make sure that you have Python 2 (`sudo apt install python`) installed on the host where you will run the synchronization script.
```

Comment:
Python 2 is EOF, python 3 should also work.

Reply:
Just verified, its working with python 3 as well

Reply:
@MeirShpilraien I tried with python3 on Ubuntu Bionic and got:
`python3 RediSearch_Syncer.py -d redis://admin:IBrS0xaL6rShfB1wKTtUkcQG1g3UWAlTd53AotPdTcQvGIVP@redis-19472.cluster1.local:19472 -s redis://admin:1GjFuUbBqTSPDbRfaxEPLSoXpFmRdFxmBKMD0BuxwMJ2DEaO@redis-19636.cluster1.local:19636
  File "RediSearch_Syncer.py", line 139
    print '\t%s:\tstatus:%s,\terrors:%s,\trepl_offset:%s' % (shards['Name'], shards['mState'], str(shards['errors']), str(shards['dst_repl_offset']))
                                                        ^
SyntaxError: invalid syntax`
That's why I put Python2.

Reply:
Its old code, try to redownload the zip, it should work now.

Reply:
ok

Reply:
It works